llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -2820,6 +2820,22 @@ SDValue DAGCombiner::visitADDLike(SDNode *N) {
   return SDValue();
 }
 
+// Attempt to form avgflooru(A, B) from add(and(A, B), lshr(xor(A, B), 1))
```

Comment:
Nit: I think the symbolic form is easier for humans to read:
```suggestion
// Attempt to form avgflooru(A, B) from (A & B) + ((X ^ B) >> 1).
```

Reply:
done.